### PR TITLE
chore(ClickableTile): add test for disabled 

### LINKED ISSUE #12750: Add tests for tile variants
Uncovered in https://github.com/carbon-design-system/carbon/pull/12701, there are some tile variants that need to have tests added.

https://github.com/carbon-design-system/carbon/issues/12750#tasklist-block-3eabe538-5a34-4391-bb7a-2091c3358a64

### PR DESCRIPTION
**Pull request by alisonjoseph**
Closes #12750

_*note, RadioTile and ExpandableTile already had tests_

#### Changelog

**New**

- add test for disabled prop on ClickableTile
- fix bug to hide onClick handler for disabled prop on ClickableTile

#### Testing / Reviewing

yarn test packages/react/src/components/Tile/Tile-test.js

### CODE REVIEW

File: packages/react/src/components/Tile/Tile-test.js [hunk 1]
```diff
@@ -56,6 +56,16 @@ describe('Tile', () => {
...
+    it('does not invoke the click handler if ClickableTile is disabled', () => {
+      const onClick = jest.fn();
+      render(
+        <ClickableTile disabled href="https://www.carbondesignsystem.com">
+          🚦
+        </ClickableTile>
+      );
+      userEvent.click(screen.getByText('🚦'));
+      expect(onClick).not.toHaveBeenCalled();
+    });
```
Comment (francinelucca) [inline packages/react/src/components/Tile/Tile-test.js]: does this need to be supplied the `onClick` function?
```suggestion
    it('does not invoke the click handler if ClickableTile is disabled', () => {
      const onClick = jest.fn();
      render(
        <ClickableTile onClick={onClick} disabled href="https://www.carbondesignsystem.com">
          🚦
        </ClickableTile>
      );
      userEvent.click(screen.getByText('🚦'));
      expect(onClick).not.toHaveBeenCalled();
    });
```

**Review by francinelucca — CHANGES_REQUESTED** [review]

**Review by andreancardona — APPROVED** [review]
Other than Francine's comment - looks good to me!

File: packages/react/src/components/Tile/Tile-test.js (hunk 1 above)
Comment (alisonjoseph) [reply packages/react/src/components/Tile/Tile-test.js]: yep thanks!

**Review by francinelucca — APPROVED** [review]
LGTM! looks like the snapshot might need to be updated @alisonjoseph

**Review by tay1orjones — APPROVED** [review]

**Pull request merged**